### PR TITLE
CI Propagation

### LINKED ISSUE #24: Automate the updating of CI on `config` branches
We do our CI changes on `main`, but we need to have those changes propagated up to the config branches as well. 

We currently do a manual `git subtree split/add` from `main` to each config branch, but this could get out of hand. 

Automate this process.

### PR DESCRIPTION
**Pull request by CodeGat**
Adding workflows to simplify future updates to CI on config branches. 

In this PR:
* Created `call-*.yml` workflows for `pr-1-ci.yml` and `pr-3-bump-tag.yml`. These workflows simply call their associated workflows that are on the main branch. 
* Updated triggers for `pr-1-ci.yml` and `pr-3-bump-tag.yml` to be `on.workflow_call`
* Updated info in `CONTRIBUTING.md` regarding how to create new config branches from scratch
* Updated `README.md` by adding a link to the above

Closes #24

### CODE REVIEW

File: CONTRIBUTING.md [hunk 1]
```diff
@@ -2,6 +2,24 @@
...
 
+### Creation of a new ACCESS-OM2 Config
+
+Config branches are entirely separate from the `main` history in this repository, save for a few files in `.github`. Note, you may need to be an Administrator to commit to `release-*` branches directly.
+
+#### If the Config is Stored in Another Repository
+
+1. Add the repository that houses the config as an upstream remote. Let's call this upstream `config`, and assume that the config is stored on the `main` branch. Let's also call the config `1deg_abc_def`.
+2. Checkout the config from one repo to this one. Run `git checkout <upstream>/<config_branch> -b release-<config_name>`. In our example, this would be `git checkout config/main -b release-1deg_abc_def`.
+3. Checkout the new branch. Run `git checkout release-<config_name>`. This would be `git checkout release-1deg_abc_def`.
```
Comment (aidanheerdegen) [inline CONTRIBUTING.md:13]: The branch is already checked out in the line above.

File: CONTRIBUTING.md [hunk 2]
```diff
@@ -2,6 +2,24 @@
...
+4. Copy across the required workflow files from the `main` branch. Run `git checkout main -- .github/workflows/call-*.yml .github/workflows/validate-json.yml`.
```
Comment (aidanheerdegen) [inline CONTRIBUTING.md:14]: Should probably be in a code block to ensure there is no confusion with the trailing full-stop.

File: CONTRIBUTING.md [hunk 3]
```diff
@@ -2,6 +2,24 @@
...
 ## Pull Request Process
 
+### Creation of a new ACCESS-OM2 Config
+
+Config branches are entirely separate from the `main` history in this repository, save for a few files in `.github`. Note, you may need to be an Administrator to commit to `release-*` branches directly.
+
+#### If the Config is Stored in Another Repository
+
+1. Add the repository that houses the config as an upstream remote. Let's call this upstream `config`, and assume that the config is stored on the `main` branch. Let's also call the config `1deg_abc_def`.
+2. Checkout the config from one repo to this one. Run `git checkout <upstream>/<config_branch> -b release-<config_name>`. In our example, this would be `git checkout config/main -b release-1deg_abc_def`.
```
Comment (aidanheerdegen) [inline CONTRIBUTING.md:12]: Should probably be in a code block to ensure there is no confusion with the trailing full-stop.

File: CONTRIBUTING.md [hunk 4]
```diff
@@ -2,6 +2,24 @@
...
+3. Checkout the new branch. Run `git checkout release-<config_name>`. This would be `git checkout release-1deg_abc_def`.
+4. Copy across the required workflow files from the `main` branch. Run `git checkout main -- .github/workflows/call-*.yml .github/workflows/validate-json.yml`.
+5. Commit and push the config branch. Due to branch protection rules, you may need to be an Administrator of the repository to commit directly to a `release-*` branch.
+
+#### If the Config is Local or New
```
Comment (aidanheerdegen) [inline CONTRIBUTING.md:17]: I don't understand that "Local" means in this context.

File: .github/workflows/call-pr-1-ci.yml [hunk 5]
```diff
@@ -0,0 +1,26 @@
...
+      - tools/**
+      - doc/**
+      - .*
+      - metadata.yaml
+      - README.md
+jobs:
+  call:
+    # We simply call the workflow on main so we don't have to propagate CI changes to
+    # multiple config branches all the time.
+    uses: codegat-test-org/configs/.github/workflows/pr-1-ci.yml@main
```
Comment (aidanheerdegen) [inline .github/workflows/call-pr-1-ci.yml:21]:
```suggestion
    uses: access-nri/access-om2-configs/.github/workflows/pr-1-ci.yml@main
```

Still pointing to your test org. Please check my suggestion is correct.

File: .github/workflows/call-pr-3-bump-tag.yml [hunk 6]
```diff
@@ -0,0 +1,19 @@
...
+  push:
+    branches:
+      - 'release-*'
+    paths:
+      - 'metadata.yaml'
+jobs:
+  call:
+    # We simply call the workflow on main so we don't have to propagate CI changes to
+    # multiple config branches all the time.
+    uses: codegat-test-org/configs/.github/workflows/pr-3-bump-tag.yml@main
```
Comment (aidanheerdegen) [inline .github/workflows/call-pr-3-bump-tag.yml:16]:
```suggestion
    uses: access-nri/access-om2-configs/.github/workflows/pr-3-bump-tag.yml@main
```

Still pointing to your test org. Better check my suggestion is correct.

**Review by aidanheerdegen — CHANGES_REQUESTED** [review]

File: CONTRIBUTING.md (hunk 4 above)
Comment (CodeGat) [reply CONTRIBUTING.md:17]: Removed that section

File: CONTRIBUTING.md (hunk 3 above)
Comment (CodeGat) [reply CONTRIBUTING.md:12]: Done

File: CONTRIBUTING.md (hunk 2 above)
Comment (CodeGat) [reply CONTRIBUTING.md:14]: Done

**Review by aidanheerdegen — APPROVED** [review]
LGTM

**Pull request merged**